core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -3072,7 +3076,7 @@ class ReplicaManager(val config: KafkaConfig,
       stateChangeLogger.info(s"Started fetchers as part of become-follower for ${partitionsToStartFetching.size} partitions")
 
       partitionsToStartFetching.foreach{ case (topicPartition, partition) =>
-        completeDelayedOperationsWhenNotPartitionLeader(topicPartition, partition.topicId)}
+        completeDelayedOperationsWhenNotPartitionLeader(topicPartition, partition.topicId, isStopReplica = false)}
```

Comment:
We can optimize this a bit more. We only need to make this call if the replica is now a follower and previously a leader.